projects/media-viewer/src/lib/annotations/annotation-set/annotation-view/annotation-view.component.ts

```diff
@@ -3,38 +3,51 @@ import { v4 as uuid } from 'uuid';
 import { Annotation } from './annotation.model';
 import { Rectangle } from './rectangle/rectangle.model';
 import { ViewerEventService } from '../../../viewers/viewer-event.service';
+import {Store} from '@ngrx/store';
+import * as fromStore from '../../../store';
+import {SelectedAnnotation} from '../../../store';
+import {SelectionAnnotation} from '../../models/event-select.model';
 
 @Component({
   selector: 'mv-annotation',
   templateUrl: './annotation-view.component.html'
 })
 export class AnnotationViewComponent {
 
-  @Input() annotation: Annotation;
+  @Input() set annotation(value) {
+      this.anno = {...value};
+  }
+  anno: Annotation;
+  selected: boolean;
   @Input() commentsLeftOffset: number;
   @Input() zoom: number;
   @Input() rotate: number;
-  @Input() selected: boolean;
+  @Input() set selectedAnnoId(selectedId) {
+    this.selected = selectedId.annotationId ? (selectedId.annotationId === this.anno.id) : false;
```

Comment:
does each annotation really need to have the selected annotation Id, should this not be in the annotationSet, and it just passes the result of this expression as an input to the annotation?

Reply:
well how it was done before the logic is was HTML so u moved it here... and i think is the best palace fo it to be for now is easier to do it hear then in level up as is using ngFor loop to pas the anno.id